lib/ansible/template/__init__.py

```diff
@@ -482,6 +499,16 @@ def template(self, variable, convert_bare=False, preserve_trailing_newlines=True
             else:
                 return variable
 
+    def is_template(self, data):
+        ''' lets us know if data has a template'''
+        try:
+            new = self.template(data)
+        except UndefinedError:
+            return True
+        except:
+            return False
+        return (new != data)
```

Comment:
pushed fix for is_template to handle complex data, before it was just passing to template which always returns either a string or guessed 'typed' result